pandas/core/arrays/categorical.py

```diff
@@ -519,6 +521,8 @@ def astype(self, dtype, copy=True):
 
         """
         if is_categorical_dtype(dtype):
+            dtype = cast(Union[str, CategoricalDtype], dtype)
```

Comment:
mypy is not able to infer the dtype type here, so I have to do a cast.

Reply:
I can add something to the contributing guide about this